streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -951,6 +956,12 @@ public StreamsMetricsImpl(Metrics metrics) {
 
             this.taskDestructionSensor = metrics.sensor(sensorNamePrefix + ".task-destruction");
             this.taskDestructionSensor.add(metrics.metricName("task-destruction-rate", metricGrpName, "The average per-second number of destructed tasks", metricTags), new Rate(new Count()));
+
+            this.processedRecordsSensor = metrics.sensor(sensorNamePrefix + ".processed-records");
```

Comment:
I see you point about the naming. However, reporting buffered record is not too helpful for the user IMHO. So not adding this sensor does make sense to me.

`Count()` is internally. Because we assume that only a few record will be dropped, to me it seems most valuable if we report the absolute number of dropped records. Or do you have a strong objection again `Count()`?